frontend/src/components/buildings/Buildings.js

```diff
@@ -1,11 +1,73 @@
 import React from 'react';
-import AddBuilding from './AddBuilding';
 import './Buildings.css';
+import { connect } from 'react-redux';
+import farm from '../../assets/buildings/farm.svg';
+import mine from '../../assets/buildings/mine.svg';
+import townhall from '../../assets/buildings/townhall.svg';
+import academy from '../../assets/buildings/barracks.svg';
+import { setErrorAction } from '../../actions/ErrorActions';
+import { getBuildingsAction} from '../../actions/BuildingsActions';
 
-export default function Buildings() {
+
+function Buildings(props){
+      
+        const buildingIcons = {
+          farm: farm,
+          mine: mine,
+          townhall: townhall,
+          academy: academy,
+        };
+      
```

Comment:
@zvorsi you can use a hook to get the data from redux, its a little bit simpler.
and you can dispatch an action to download the buildings when loaded
```suggestion
      const buildings = useSelector(state => state.buildings)
      const dispatch = useDispatch()
      useEffect(() => {
        dispatch(getBuildingsAction)
      }, [])
```